subsys/net/l2/openthread/openthread.c

```diff
@@ -187,26 +187,20 @@ static void ot_state_changed_handler(uint32_t flags, void *context)
 {
 	struct openthread_state_changed_cb *entry, *next;
 	struct openthread_context *ot_context = context;
+	bool isUp = otIp6IsEnabled(ot_context->instance);
 
-	NET_INFO("State changed! Flags: 0x%08" PRIx32 " Current role: %s",
+	NET_INFO("State changed! Flags: 0x%08" PRIx32 " Current role: %s" " Ip6: %s",
 		flags,
-		otThreadDeviceRoleToString(otThreadGetDeviceRole(ot_context->instance))
-		);
-
-	if (flags & OT_CHANGED_THREAD_ROLE) {
-		switch (otThreadGetDeviceRole(ot_context->instance)) {
-		case OT_DEVICE_ROLE_CHILD:
-		case OT_DEVICE_ROLE_ROUTER:
-		case OT_DEVICE_ROLE_LEADER:
-			net_if_dormant_off(ot_context->iface);
-			break;
+		otThreadDeviceRoleToString(otThreadGetDeviceRole(ot_context->instance)),
+		(isUp ? "up" : "down"));
 
-		case OT_DEVICE_ROLE_DISABLED:
-		case OT_DEVICE_ROLE_DETACHED:
-		default:
-			net_if_dormant_on(ot_context->iface);
-			break;
-		}
+	if (isUp)
+	{
+		net_if_dormant_off(ot_context->iface);
+	}
+	else
+	{
+		net_if_dormant_on(ot_context->iface);
```

Comment:
Curly braces for `if` statements should not go on a new line.

```suggestion
	if (isUp) {
		net_if_dormant_off(ot_context->iface);
	} else {
		net_if_dormant_on(ot_context->iface);
```